### PR TITLE
Cv 1105 dataloader update

### PR DESCRIPTION
**Pull request by anandravi24**
Hello Guys,
This is a small PR to combine repetitive sections of the dataloader script.
One of the TODO was to merge the repetitive sections present in the dataloader script.
Using inheritance, I have managed to trim down the repetitive section mostly present in __getitem__(). 

Major Changes:
Introduced inheritance for code reusing
1) Added TrainValBaseLoader class which will be inherited by TrainSet Loader and Val Set loader
2) Trimmed TestLoadercompare by inheriting major functionality from TestLoader

### CODE REVIEW

File: dataloader.py [hunk 1]
```diff
@@ -91,49 +85,52 @@ def __init__(self, dataset_dir, cfg, val_ids):
...
+            self.dataset_dir + '/lr_x' + str(self.scale_factor) + '/' + self.file_list[index] + '/lr0.png')
+        lr_image_right = Image.open(
+            self.dataset_dir + '/lr_x' + str(self.scale_factor) + '/' + self.file_list[index] + '/lr1.png')
+
         hr_image_left = ToTensor()(hr_image_left)
         hr_image_right = ToTensor()(hr_image_right)
         lr_image_left = ToTensor()(lr_image_left)
         lr_image_right = ToTensor()(lr_image_right)
-        return hr_image_left, hr_image_right, lr_image_left, lr_image_right, self.file_list[index]
+        return hr_image_left, hr_image_right, lr_image_left, lr_image_right
```
Comment (OwenHua666) [inline dataloader.py:117]: Can we also reuse the base get_item here?

Comment (anandravi24) [reply dataloader.py:117]: Hey Owen, I didnt reuse the get_item here because the test sets have different file directory system in which the data is stored. So the test loaders have different getItem and train and val loader have different getitem

Comment (OwenHua666) [reply dataloader.py:117]: Cool. Can I ask why the file system is different?

Comment (anandravi24) [reply dataloader.py:117]: The trainset and test set are generated using the matlab files ```generate_trainset.m``` and ```generate_testset.m```. These two files are provided by the authors of the PASSRnet. They store the data in a different folder system
